### PR TITLE
feat: [CDS-82572]: show git cache status icon for webhooks in service and environments details page

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Show Git cache status in service / environment details page and added support for reload from git

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/8bd9243b-85d6-40ab-9d17-c8fc03eb27c0

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsHeader/ServiceDetailsHeader.tsx [hunk 1]
```diff
@@ -41,13 +41,18 @@ import { StoreType } from '@common/constants/GitSyncTypes'
 import notificationImg from './notificationImg.svg'
 import css from '@cd/components/ServiceDetails/ServiceDetailsHeader/ServiceDetailsHeader.module.scss'
 
+interface ServiceDetailHeaderProps {
+  handleReloadFromCache: () => void
+}
+
 export const ServiceDetailsHeader = (
-  _props: unknown,
+  _props: ServiceDetailHeaderProps,
```
Comment (PratyushGar) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsHeader/ServiceDetailsHeader.tsx:49]: nit:
This can be `props: ServiceDetailHeaderProps,`

Comment (vikrant-harness) [reply src/modules/75-cd/components/ServiceDetails/ServiceDetailsHeader/ServiceDetailsHeader.tsx]: done

**Review by sunnykesh-harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]
Meaningful test case please.

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Comment by vikrant-harness** [comment]
Automation ticket :- https://harness.atlassian.net/browse/CDS-83045

**Pull request merged**